code/modules/projectiles/mounted.dm

```diff
@@ -0,0 +1,282 @@
+	. = ..()
+	gun.examine(user)
+
+/obj/machinery/mounted/attackby(obj/item/I, mob/user, params) //This handles reloading the gun, if its in acid cant touch it.
+	. = ..()
+
+	if(!ishuman(user))
+		return
+
+	for(var/obj/effect/xenomorph/acid/A in loc)
+		if(A.acid_t == src)
+			to_chat(user, "You can't get near that, it's melting!")
+			return
+
+	reload(user, I)
+
+///Repairs gun
+/obj/machinery/mounted/welder_act(mob/living/user, obj/item/I)
+	if(user.do_actions)
+		return FALSE
+
+	var/obj/item/tool/weldingtool/WT = I
+
+	if(!WT.isOn())
+		return FALSE
+
+	for(var/obj/effect/xenomorph/acid/A in loc)
+		if(A.acid_t == src)
+			to_chat(user, "You can't get near that, it's melting!")
+			return TRUE
+
+	if(obj_integrity == max_integrity)
+		to_chat(user, "<span class='warning'>[src] doesn't need repairs.</span>")
+		return TRUE
+
+	if(user.skills.getRating("engineer") < SKILL_ENGINEER_METAL)
+		user.visible_message("<span class='notice'>[user] fumbles around figuring out how to repair [src].</span>",
+		"<span class='notice'>You fumble around figuring out how to repair [src].</span>")
+		var/fumbling_time = 5 SECONDS * ( SKILL_ENGINEER_METAL - user.skills.getRating("engineer") )
+		if(!do_after(user, fumbling_time, TRUE, src, BUSY_ICON_BUILD))
+			return TRUE
+
+	user.visible_message("<span class='notice'>[user] begins repairing damage to [src].</span>",
+	"<span class='notice'>You begin repairing the damage to [src].</span>")
+	playsound(loc, 'sound/items/welder2.ogg', 25, TRUE)
+
+	if(!do_after(user, 5 SECONDS, TRUE, src, BUSY_ICON_FRIENDLY))
+		return TRUE
+
+	if(!WT.remove_fuel(2, user))
+		to_chat(user, "<span class='warning'>Not enough fuel to finish the task.</span>")
+		return TRUE
+
+	user.visible_message("<span class='notice'>[user] repairs some damage on [src].</span>",
+	"<span class='notice'>You repair [src].</span>")
+	playsound(loc, 'sound/items/welder2.ogg', 25, TRUE)
+	repair_damage(120)
+	update_icon_state()
+	return TRUE
+
+///Reloads gun
+/obj/machinery/mounted/proc/reload(mob/user, ammo_magazine)
+	if(!istype(ammo_magazine, /obj/item/ammo_magazine))
+		return
+
+	var/obj/item/ammo_magazine/ammo = ammo_magazine
+	
+	if(istype(gun, ammo.gun_type))
+
+		if(ammo.current_rounds <= 0)
+			to_chat(user, "<span class='warning'>[ammo] is empty!</span>")
+			return
+
+		if(gun.current_mag)
+			gun.unload(user,0,1)
+			update_icon_state()
+
+		var/tac_reload_time = max(0.5 SECONDS, 1.5 SECONDS - user.skills.getRating("firearms") * 5)
+		if(!do_after(user, tac_reload_time, TRUE, ammo_magazine))
+			return
+	
+		gun.reload(user, ammo_magazine)
+		update_icon_state()
+
+///Sets the user as manning the internal gun
+/obj/machinery/mounted/on_set_interaction(mob/user)
+	. = ..()
+	operator = user
+	update_view(operator)
+	gun.manned(operator)
+
+///Unsets the user from manning the internal gun
+/obj/machinery/mounted/on_unset_interaction(mob/user)
+	if(user.client)
+		user.client.change_view(WORLD_VIEW)
+		user.client.pixel_x = 0
```

Comment:
note this is being changed soon in datumized view

Reply:
thank god